### PR TITLE
feat: Refactor and add tests for load-mock

### PR DESCRIPTION
**Pull request by markstory**
Refactor load-mocks into a set of smaller functions, and have some high level tests to ensure that load-mocks will run. These tests aren't exhaustive but they should help us catch and prevent regressions in load-mocks in the future.

### CODE REVIEW

**Review by corps — APPROVED** [review]
The diff is hard to follow, but that's understandable, it has no tests and is a dev tool.  So I'm taking your word that this works locally, I do think the approach is good 👍

**Comment by markstory** [comment]
Ran this locally before merging and it runs cleanly.
